ee/localserver/krypto-ec-middleware.go

```diff
@@ -47,18 +50,27 @@ func newKryptoEcMiddleware(logger log.Logger, localDbSigner, hardwareSigner cryp
 
 func (e *kryptoEcMiddleware) Wrap(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx, span := traces.New(r.Context())
+		defer span.End()
```

Comment:
This is an interesting question... I don't know if we should add spans to `kryptoEcMiddleware.Wrap` vs wrapping with a span middleware. Thoughts?

Reply:
I thought it was helpful this way because it gives us more specific timing info for the `kryptoEcMiddleware`, which would've been helpful in the past. We do already effectively wrap the entire localhost handler with the span middleware -- we use `otelhttp.NewHandler` to wrap all our other handlers. So we have an outer span for the initial request, a child span for the kryptoEcMiddleware, and then a final child span for the request handling (e.g. control server acceleration or whatever we're doing).

Reply:
That's a good point -- maybe we don't _usually_ want to instrument a span, but this one might be special.

Reply:
There were a couple aspects to the original comment, but I'm not sure they're germane. 

Foremost, my point is about how if we want to instrument the entire span, we'd be better off doing it in middleware. Though you're correct that `otelhttp.NewHandler` is already that middleware.

The secondary point is that if we're instrumenting the span  with `otelhttp.NewHandler`, then immediately starting a subspan isn't going to be meaningful. It should be the same timing period. If there are smaller parts we want to trace it makes sense, but this looks like:
```
trace1{
  trace2{
    stuff
  }
}
```

So I don't see it adding much

Reply:
Though I see that you need the `span` to add in the error. But I wonder if `SpanFromContext` is a better fit. 

Basically, why add the span? Does it help any?

Reply:
In general I like more child spans because they help us narrow down causes of performance issues. E.g. here, if we have a localserver request that we see take a long time, we can use the multiple child spans to identify if it's the krypto middleware that's taking a while, or the inner request to e.g. accelerate the control server interval. Without the child spans, all we have is one span that tells us it takes a certain amount of time for the request to be processed and a response to be returned, which isn't really more info than what we've got in the browser dev console.

Reply:
Right. But only if there are those inner spans. So why wrap the whole function, vs just the blocks you want? There's an extra span here

Reply:
I think we're talking past each other. But we can point at a screen with real data and iterate later